runtimevar/paramstore/paramstore.go

```diff
@@ -27,18 +34,107 @@ package paramstore // import "gocloud.dev/runtimevar/paramstore"
 import (
 	"context"
 	"fmt"
+	"net/url"
+	"path"
+	"sync"
 	"time"
 
+	gcaws "gocloud.dev/aws"
 	"gocloud.dev/gcerrors"
 	"gocloud.dev/runtimevar"
 	"gocloud.dev/runtimevar/driver"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ssm"
 )
 
+func init() {
+	runtimevar.DefaultURLMux().RegisterVariable(Scheme, new(lazySessionOpener))
+}
+
+// URLOpener opens AWS Paramstore URLs like "paramstore://myvar".
+// See gocloud.dev/aws/ConfigFromURLParams for supported query parameters
+// that affect the default AWS session.
+//
+// In addition, the following URL parameters are supported:
+//   - decoder: The decoder to use. Defaults to URLOpener.Decoder, or
+//       runtimevar.BytesDecoder if URLOpener.Decoder is nil.
+//       See runtimevar.DecoderByName for supported values.
+//   - wait: The poll interval; supported values are from time.ParseDuration.
+//       Defaults to 30s.
+type URLOpener struct {
+	// ConfigProvider must be set to a non-nil value.
+	ConfigProvider client.ConfigProvider
+
+	// Decoder and Options can be specified at URLOpener construction time,
+	// or provided/overridden via URL parameters.
+	Decoder *runtimevar.Decoder
+	Options Options
+}
+
+// lazySessionOpener obtains the AWS session from the environment on the first
+// call to OpenVariableURL.
+type lazySessionOpener struct {
+	init   sync.Once
+	opener *URLOpener
+	err    error
+}
+
+func (o *lazySessionOpener) OpenVariableURL(ctx context.Context, u *url.URL) (*runtimevar.Variable, error) {
+	o.init.Do(func() {
+		sess, err := session.NewSessionWithOptions(session.Options{SharedConfigState: session.SharedConfigEnable})
```

Comment:
For AWS do we not need to provide region? Is it included in the `SharedConfigEnable`?

Reply:
`SharedConfigEnable` means it will get a default value from your local environment (e.g., `aws login`). So, maybe.

We'll get the region parameter from the URL down below, see `gcaws.ConfigFromURLParams(q)`, and layer it on top of this before connecting.